gendoc/src/main.rs

```diff
@@ -74,17 +75,58 @@ FROM pg_catalog.pg_operator o
 ORDER BY schema_visibility.public DESC, schema, name, left_arg_type, right_arg_type;
 "#;
 
-fn main() {
+fn generate_random_db_name() -> String {
+    let mut rng = rand::thread_rng();
+    let n2: u16 = rng.gen();
+    format!("test_database_{}", n2)
+}
+
+fn in_temporary_local_db<F>(cb: F)
+where
+    F: Fn(&mut Client),
+{
+    let db_name = generate_random_db_name();
+    let mut admin_conn = connect_local("template1");
+    admin_conn
+        .simple_query(format!("CREATE DATABASE {};", db_name).as_str())
+        .unwrap();
+    let mut db_conn = connect_local(db_name.as_str());
+    cb(&mut db_conn);
+    drop(db_conn);
+    admin_conn
+        .simple_query(format!("DROP DATABASE {};", db_name).as_str())
+        .unwrap();
+}
+
+fn in_docker_db<F>(cb: F)
+where
+    F: Fn(&mut Client),
+{
     let docker = clients::Cli::default();
     let node = run_postgres(&docker);
-    let mut connection = connect(&node);
+    let mut connection = connect_docker(&node);
+    cb(&mut connection);
+}
+
+fn main() {
+    let use_local = env::var("GENDOC_USE_LOCAL")
+        .map(|val| val.to_ascii_lowercase() == "true")
+        .unwrap_or(false);
```

Comment:
I would prefer this to default to `true`

Reply:
Done.